File: csharp/Exo3/PersonClient.cs
using System.IO;

namespace Soat.CleanCoders.Refactoring.Exo3
{
    public class PersonClient
    {
        public void Client1(TextWriter @out, Person person)
        {
            @out.Write(person.first);
            @out.Write(" ");
            if (!string.IsNullOrEmpty(person.middle))
            {
                @out.Write(person.middle);
                @out.Write(" ");
            }

            @out.Write(person.last);
        }

        public string Client2(Person person)
        {
            var result = person.last + ", " + person.first;
            if (!string.IsNullOrEmpty(person.middle))
            {
                result += " " + person.middle;
            }

            return result;
        }

        public void Client3(TextWriter @out, Person person)
        {
            @out.Write(person.last);
            @out.Write(", ");
            @out.Write(person.first);
            if (!string.IsNullOrEmpty(person.middle))
            {
                @out.Write(" ");
                @out.Write(person.middle);
            }
        }

        public string Client4(Person person)
        {
            return person.last + ", " + person.first + (string.IsNullOrEmpty(person.middle)
                                                            ? ""
                                                            : " " + person.middle);
        }
    }
}

File: javascript/app/exo3/person-client.js
function PersonClient() {
	
	this.client1 = client1;
	this.client2 = client2;
	this.client3 = client3;
	this.client4 = client4;
	
	function client1(out, person) {
		out.log(person.first);
		out.log(" ");
		if (person.middle != null) {
			out.log(person.middle);
			out.log(" ");
		}
		out.log(person.last);
	}
	
	function client2(person) {
		let result = person.last + ", " + person.first;
		if (person.middle != null)
			result += " " + person.middle;
		return result;
	}
	
	function client3(out, person) {
		out.log(person.last);
		out.log(", ");
		out.log(person.first);
		if (person.middle != null) {
			out.log(" ");
			out.log(person.middle);
		}
	}
	
	function client4(person) {
		return person.last + ", " +
			person.first +
			((person.middle == null) ? "" : " " + person.middle);
	}
}

module.exports = PersonClient;